pandas/core/frame.py

```diff
@@ -1266,7 +1266,7 @@ def from_dict(cls, data, orient="columns", dtype=None, columns=None) -> "DataFra
             if len(data) > 0:
                 # TODO speed up Series case
                 if isinstance(list(data.values())[0], (Series, dict)):
```

Comment:
Not your PR, but this line above is potentially very expensive. Can you change it to:

```python
first_val = next(iter((data.values())), None)
if isinstance(first_val, (Series, dict)):
```

to avoid creating that list. Does this make a difference in your ASVs?

Reply:
Nice! the ASV results are even better:

```
       before           after         ratio
     [c6c53671]       [bcb25b91]
     <master>         <TODO-cythonize>
-      1.03±0.1ms          748±6μs     0.72  series_methods.NanOps.time_func('sum', 1000000, 'int8')

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
PERFORMANCE INCREASED.
```